[excel] (Events) Article section and samples for the onRowSorted and onColumnSorted events (#1298)
OfficeDev/office-js-docs-pr: docs/excel/excel-add-ins-worksheets.md

}
```

> ## Sorting events (preview)

Reviewer: ```suggestion
## Handle sort-related events (preview)
```

Author: @kbrandl, How do you feel about calling this "Handle sorting events"? The "related" part makes the events seem distant to the actual sorting.

Reviewer: That sounds good to me! Be sure to update the hyperlink (text and URL) in the other article that points to this section.

> [!NOTE]
> The APIs for these sort-related events are currently available only in public preview. [!INCLUDE [Information about using preview APIs](../includes/using-excel-preview-apis.md)]

> The `onColumnSorted` and `onRowSorted` events indicate when any worksheet data is sorted. These events fire whether the sorting is done through UI controls (like table drop-downs) or programmatically. These events are connected to individual `Worksheet` objects and the workbook's `WorkbookCollection`.

Reviewer: ```suggestion
The `onColumnSorted` and `onRowSorted` events indicate when any worksheet data is sorted. These events are connected to individual `Worksheet` objects and to the workbook's `WorkbookCollection`, and will fire whether the sorting is done manually through the Excel user interface or programmatically.
```

> [!NOTE]

> > `onColumnSorted` fires when columns are sorted as the result of a left-to-right sort operation. `onRowSorted` fires when rows are sorted as the result of a top-to-bottom sort operation. Sorting a table using the drop-down on a column header results in an `onRowSorted` event. The event corresponds with what is moving, not what is being considered as the sorting criteria.

Reviewer: The style guide won't load for me right now, but IIRC, "drop-down" isn't quite the right terminology. Suggest you look up the recommended term in the style guide (if it'll load for you).

Author: I double-checked and drop-down should be okay for developers.

![image](https://user-images.githubusercontent.com/38896772/64637736-44e85c80-d3b9-11e9-850e-34ae087bf9c6.png)

Author: Though in re-reading that, I'll need to clarify the control type. I'll figure out how best to describe that.

Reviewer 2: I think if you choose to use that term, then it might be "dropdown"

Author: I think the best way to describe that Excel control is as a "drop-down menu", like in the above screenshot.

The `onColumnSorted` and `onRowSorted` events provide their callbacks with [WorksheetColumnSortedEventArgs](/javascript/api/excel/excel.worksheetcolumnsortedeventargs) or [WorksheetRowSortedEventArgs](/javascript/api/excel/excel.worksheetrowsortedeventargs), respectively. These give more details about the event. In particular, both `EventArgs` have an `address` property that represents the rows or columns moved as a result of the sort operation. Any cell with sorted content is included, even if that cell's value was not part of the sorting criteria.

> The following images show what ranges are returned by the `address` property for sort events. First, here is the sample data before sorting:

Reviewer: ```suggestion
The following images show the ranges returned by the `address` property for sort events. First, here is the sample data before sorting:
```

![Table data in Excel before being sorted](../images/excel-sort-event-before.png)

If a top-to-bottom sort is performed on "**Q1**" (the values in "**B**"), the following highlighted rows are returned by `WorksheetRowSortedEventArgs.address`:

![Table data in Excel after a top-to-bottom sort. The rows that have moved are highlighted.](../images/excel-sort-event-after-row.png)

If a left-to-right sort is performed on "**Quinces**" (the values in "**4**") on the original data, the following highlighted columns are returned by `WorksheetColumnsSortedEventArgs.address`:

![Table data in Excel after a left-to-right sort. The columns that have moved are highlighted.](../images/excel-sort-event-after-column.png)

> The following code sample shows how to add an event handler to `Worksheet.onRowSorted`. The handler's callback clears the fill color for the range, then fills the cells of the moved rows.

Reviewer: ```suggestion
The following code sample shows how to register an event handler for the `Worksheet.onRowSorted` event. The handler's callback clears the fill color for the range, then fills the cells of the moved rows.
```

> ```js

Reviewer: Should the "fat arrow" syntax be used in this (JavaScript) code sample, or should the longer `function (...)` syntax be used instead?
